api/include/opentelemetry/http/http_client.h

```diff
@@ -0,0 +1,161 @@
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#pragma once
+
+#include "opentelemetry/nostd/string_view.h"
+
+OPENTELEMETRY_BEGIN_NAMESPACE
+namespace http
+{
+
+enum class HttpResult : uint8_t
+{
+  // Response has been received successfully from target server.
+  HttpResult_OK = 0,
+
+  // Request has been aborted by the caller. The server might or
+  // might not have already received or processed the request.
+  HttpResult_Aborted = 1,
+
+  // Local conditions have prevented the request from being sent
+  // (invalid request parameters, out of memory, internal error etc.)
+  HttpResult_LocalFailure = 2,
+
+  // Network conditions somewhere between the local machine and
+  // the target server have caused the request to fail
+  // (connection failed, connection dropped abruptly etc.).
+  HttpResult_NetworkFailure = 3
+
+};
+
+// The HttpHeaders class contains a set of HTTP headers.
+class HttpHeaders
```

Comment:
Is there a place where do use `HttpHeaders` in API? Does it mean we need to have a transform somewhere from `std::multimap<std::string, std::string>` used within the SDK - to ABI-safe `HttpHeaders` implementation used in API surface? Can you provide some concrete example where we require this in API?

Reply:
I don't see a use case for this in the API.